Migrating from RequireJS to ES6 Modules

farhan opened the pull request:
Ticket: https://github.com/openedx/edx-platform/issues/36413

- In this PR we are moving Video Block JS into [xmodule/assets](https://github.com/openedx/edx-platform/tree/master/xmodule/assets) as discussed in this [readme](https://github.com/openedx/edx-platform/tree/master/xmodule/assets#readme)
- Replacing RequireJS with ES6 `import/export`

### JS files refactoring details:

- Reviewer can open the second or next commits to review the changes as the first commit contains the old code.
- [diffchecker](https://www.diffchecker.com/text-compare/) can be used (on real-time-diff) to see the accurate refactoring changes.

| #  | Source JS File (RequireJS pattern) | Destination JS File (Vanilla JS) |
|----|------------------------------------|-----------------------------------|
| 1  | [00_i18n.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/00_i18n.js) | [00_i18n.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/00_i18n.js) |
| 2  | [00_async_process.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/00_async_process.js) | [00_async_process.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/00_async_process.js) |
| 3  | [09_poster.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/09_poster.js) | [09_poster.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/09_poster.js) |
| 4  | [035_video_accessible_menu.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/035_video_accessible_menu.js) | [035_video_accessible_menu.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/035_video_accessible_menu.js) |
| 5  | [00_iterator.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/00_iterator.js) | [00_iterator.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/00_iterator.js) |
| 6  | [09_skip_control.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/09_skip_control.js) | [09_skip_control.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/09_skip_control.js) |
| 7  | [09_play_placeholder.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/09_play_placeholder.js) | [09_play_placeholder.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/09_play_placeholder.js) |
| 8  | [00_component.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/00_component.js) | [00_component.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/00_component.js) |
| 9  | [09_play_skip_control.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/09_play_skip_control.js) | [09_play_skip_control.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/09_play_skip_control.js) |
| 10 | [09_play_pause_control.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/09_play_pause_control.js) | [09_play_pause_control.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/09_play_pause_control.js) |
| 11 | [036_video_social_sharing.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/036_video_social_sharing.js) | [036_video_social_sharing.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/036_video_social_sharing.js) |
| 12 | [00_video_storage.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/00_video_storage.js) | [00_video_storage.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/00_video_storage.js) |
| 13 | [10_commands.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/10_commands.js) | [10_commands.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/10_commands.js) |
| 14 | [09_bumper.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/09_bumper.js) | [09_bumper.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/09_bumper.js) |
| 15 | [09_events_bumper_plugin.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/09_events_bumper_plugin.js) | [09_events_bumper_plugin.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/09_events_bumper_plugin.js) |
| 16 | [00_sjson.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/00_sjson.js) | [00_sjson.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/00_sjson.js) |
| 17 | [08_video_auto_advance_control.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/08_video_auto_advance_control.js) | [08_video_auto_advance_control.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/08_video_auto_advance_control.js) |
| 18 | [09_save_state_plugin.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/09_save_state_plugin.js) | [09_save_state_plugin.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/09_save_state_plugin.js) |
| 19 | [025_focus_grabber.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/025_focus_grabber.js) | [025_focus_grabber.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/025_focus_grabber.js) |
| 20 | [02_html5_hls_video.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/02_html5_hls_video.js) | [02_html5_hls_video.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/02_html5_hls_video.js) |
| 21 | [09_events_plugin.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/09_events_plugin.js) | [09_events_plugin.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/09_events_plugin.js) |
| 22 | [04_video_control.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/04_video_control.js) | [04_video_control.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/04_video_control.js) |
| 23 | [05_video_quality_control.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/05_video_quality_control.js) | [05_video_quality_control.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/05_video_quality_control.js) |
| 24 | [00_resizer.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/00_resizer.js) | [00_resizer.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/00_resizer.js) |
| 25 | [09_completion.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/09_completion.js) | [09_completion.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/09_completion.js) |
| 26 | [10_main.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/10_main.js) | [10_main.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/10_main.js) |
| 27 | [037_video_transcript_feedback.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/037_video_transcript_feedback.js) | [037_video_transcript_feedback.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/037_video_transcript_feedback.js) |
| 28 | [04_video_full_screen.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/04_video_full_screen.js) | [04_video_full_screen.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/04_video_full_screen.js) |
| 29 | [06_video_progress_slider.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/06_video_progress_slider.js) | [06_video_progress_slider.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/06_video_progress_slider.js) |
| 30 | [02_html5_video.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/02_html5_video.js) | [02_html5_video.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/02_html5_video.js) |
| 31 | [08_video_speed_control.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/08_video_speed_control.js) | [08_video_speed_control.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/08_video_speed_control.js) |
| 32 | [07_video_volume_control.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/07_video_volume_control.js) | [07_video_volume_control.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/07_video_volume_control.js) |
| 33 | [095_video_context_menu.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/095_video_context_menu.js) | [095_video_context_menu.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/095_video_context_menu.js) |
| 34 | [01_initialize.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/01_initialize.js) | [01_initialize.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/01_initialize.js) |
| 35 | [03_video_player.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/03_video_player.js) | [03_video_player.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/03_video_player.js) |
| 36 | [09_video_caption.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/09_video_caption.js) | [09_video_caption.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/09_video_caption.js) |
| 37  | [time.js](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/time.js) | [time.js](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/utils/time.js) |


### How to test:
- Check out the branch
- run `npm run webpack-dev`
- Do Hard cache reload of the browser


### Old archived [PR](https://github.com/openedx/edx-platform/pull/36527)

farhan committed:
chore: Original JS files copied from xmodule/js/src/video directory

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9925500016-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9925500016-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9925500016-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9926325594-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9926325594-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9926325594-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9927665395-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9927665395-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9927665395-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9928050964-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9928050964-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9928050964-tutor-requirements.txt)

bradenmacdonald commented:
Wow, great to see this. Question: why are the old .js files not deleted as part of this PR?

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9939157206.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9939157206-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9939157206-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9939157206-tutor-requirements.txt)

ahmad-arbisoft commented on xmodule/video_block/video_block.py:
```diff
@@ -280,8 +280,8 @@ def public_view(self, context):
-        add_webpack_js_to_fragment(fragment, 'VideoBlockDisplay')
-        shim_xmodule_js(fragment, 'Video')
+        add_webpack_js_to_fragment(fragment, 'VideoBlockMain')
+        fragment.initialize_js('Video')
```
What is the reason for this?

farhan replied:
It will call [this](https://github.com/openedx/edx-platform/pull/36595/files#diff-4abb88de787a590bc61d35d3438752f3507a2983d5614216e13d031fffe54d13R42) Video function define in the relevant JS file

ahmad-arbisoft commented on xmodule/video_block/video_block.py:
```diff
@@ -246,8 +246,8 @@ def student_view(self, context):
-        add_webpack_js_to_fragment(fragment, 'VideoBlockDisplay')
-        shim_xmodule_js(fragment, 'Video')
+        add_webpack_js_to_fragment(fragment, 'VideoBlockMain')
+        fragment.initialize_js('Video')
```
What is the reason for this?

farhan replied:
same reason

ahmad-arbisoft commented on xmodule/assets/video/public/js/utils/time.js:
```diff
@@ -0,0 +1,41 @@
+    minutes %= 60;
+
+    if (formatFull) {
+        return '' + _pad(hours) + ':' + _pad(minutes) + ':' + _pad(seconds % 60);
```
What is _pad doing here? is this a variable or something from imports?

farhan replied:
It's a function defined at end of file
https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/utils/time.js#L35

Whole file has been moved [from](https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/time.js)

kdmccormick replied:
Can we delete the xmodule/js/src/time.js now, or is it still in use by other builtin xblock JS?

ahmad-arbisoft commented on xmodule/assets/video/public/js/09_skip_control.js:
```diff
@@ -0,0 +1,72 @@
+
+    /** Initializes the module. */
+    initialize: function() {
+        this.el = $(this.template);
```
I am not sure of this, what if we dont have `this.template` or have it as null?

farhan replied:
template is define above [here](https://github.com/openedx/edx-platform/blob/farhan/move-video-js-to-assets/xmodule/assets/video/public/js/09_play_skip_control.js#L26), so don't think it could be null

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9945897449-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9945897449-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9945897449-tutor-requirements.txt)

ahmad-arbisoft reviewed (approved):
LGTM

farhan commented:
> Wow, great to see this. Question: why are the old .js files not deleted as part of this PR?

@bradenmacdonald 
Thanks!
I was thinking to delete old JS files in separate PR, I think it makes better sense to delete them within this PR.
On the way.

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9964248789-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9964248789-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-9964248789-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10004875524-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10004875524-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10004875524-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10016112722.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10016112722-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10016112722-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10016112722-tutor-requirements.txt)

ttqureshi commented on xmodule/assets/video/public/js/03_video_player.js:
```diff
@@ -1,914 +1,912 @@
+import _ from 'underscore';
+import { convert, format, formatFull } from './utils/time.js';
+
+let dfd = $.Deferred();
```
Do we need to import `$` as done in other files? 
`import $ from 'jquery';`?

ttqureshi commented on xmodule/assets/video/public/js/03_video_player.js:
```diff
@@ -1,914 +1,912 @@
+    videoPlayer.endTime = this.config.endTime;
+    if (
+        videoPlayer.endTime <= videoPlayer.startTime
+&& videoPlayer.endTime >= duration
```
are we sure about this change? `||` -> `&&`

farhan replied:
Good point it, reverting

ttqureshi reviewed (approved):
LGTM

kdmccormick reviewed (commented):
Overall-- Excellent work. 

I am performing a detailed line-by-line review currently, which will take some time, but I don't expect there to be any major change requests.

Also, thank you splitting this into 2 commits, that has been very helpful. By the way, I have been able to review the diff easily on GitHub by selecting "Hide whitespace" which hides away all the indentation/spacing changes.

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10016112703-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10016112703-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10016112703-tutor-requirements.txt)

kdmccormick commented on xmodule/assets/video/public/js/01_initialize.js:
```diff
@@ -11,835 +11,835 @@
+
+function getDuration() {
+    try {
+        let safeMoment = typeof moment !== 'undefined' ? moment : window.moment;
```
I see that if we import `moment` and it is undefined, then we fall back to `window.moment`. Can you explain why `moment` in particular might be undefined after import?

farhan replied:
Actually I don't know.

I refactored the logic from here:
https://github.com/openedx/edx-platform/blob/master/xmodule/js/src/video/01_initialize.js#L19

Perhaps we should put it at top of file
or can remove the null or undefined check if we are confident about it.

kdmccormick replied:
Ah I didn't see that, I thought was new logic. Since it is existing logic I think it's OK to leave as-is.

kdmccormick commented on xmodule/assets/video/public/js/037_video_transcript_feedback.js:
```diff
@@ -1,247 +1,241 @@
+ * @access public
+ *
+ * @param {object} state - The object containing the state of the video
+ *     player. All other modules, their parameters, public letiables, etc.
```
"letiables" 😄

farhan replied:
😆  Bad find and replace 
reverted.

kdmccormick commented on xmodule/assets/video/public/js/037_video_transcript_feedback.js:
```diff
@@ -1,247 +1,241 @@
+import _ from 'underscore';
+
+/**
+ * @desc VideoTranscriptFeedbackHandler module exports a function.
```
VideoTranscriptFeedbackHandler in this doc comment should be updated to match how you renamed the class

kdmccormick commented on xmodule/assets/video/public/js/037_video_transcript_feedback.js:
```diff
@@ -1,247 +1,241 @@
+    }
+};
+
+// Export as ES6 default
```
Unnecessary comment -- this is just explaining how javascript works

kdmccormick commented on xmodule/assets/video/public/js/037_video_transcript_feedback.js:
```diff
@@ -1,247 +1,241 @@
+};
+
+// Export as ES6 default
+export default VideoTranscriptFeedback;
```
missing newline at the end of the file

kdmccormick commented on xmodule/assets/video/public/js/03_video_player.js:
```diff
@@ -1,914 +1,912 @@
+    return dfd.promise();
+};
+
+/* eslint-disable no-use-before-define */
```
since you've added the `let`, this eslint-disable comment can be removed.

kdmccormick commented on webpack.builtinblocks.config.js:
```diff
@@ -79,10 +79,6 @@ module.exports = {
             './xmodule/js/src/xmodule.js',
             './xmodule/js/src/sequence/edit.js'
         ],
-        VideoBlockDisplay: [
-            './xmodule/js/src/xmodule.js',
-            './xmodule/js/src/video/10_main.js'
-        ],
```
I don't see a reason to move this to webpack.common.config.js or to change the name of the entrypoint. After all, Video is still a built-in block (for now). Would it be work to change this to:
```js
        VideoBlockDisplay: [
            './xmodule/js/src/video/10_main.js'
        ],
```

farhan replied:
Yes, its working
I was actually trying to make this block configs similar to vertical block as it was mentioned as an example in the readme [here](https://github.com/openedx/edx-platform/blob/master/xmodule/assets/README.rst#javascript-js)
but I think it make more sense as you are suggesting.

kdmccormick commented on xmodule/assets/video/public/js/10_main.js:
```diff
@@ -1,198 +1,147 @@
+if (typeof window.Logger === 'undefined') {
+    window.Logger = function(eventName, data) {
+        console.log(`Event: ${eventName}, Data: ${data}`);
+    }
```
When does this happen? Did we need this gettext fallback logic before, and if not, why do we need it now?

farhan replied:
I have removed both `gettext` and `Logger` fallback methods.
Actually there were required while development, now code is working fine without them so removed them now.

kdmccormick reviewed (changes requested):
Just some minor questions and requests.

kdmccormick reviewed (commented):
Code looks great, ty!

I'm just performing some final manual testing and then I will approve.

I've messaged #cc-risky-changes that we plan to merge tomorrow.

farhan commented:
> Code looks great, ty!
> 
> I'm just performing some final manual testing and then I will approve.
> 
> I've messaged #cc-risky-changes that we plan to merge tomorrow.

@kdmccormick Thanks,

One important thing I want to discuss is the [karma_runner_webpack](https://github.com/openedx/edx-platform/blob/master/xmodule/js/karma_runner_webpack.js#L40) tests code which exists in [directory](https://github.com/openedx/edx-platform/tree/master/xmodule/js/spec)
 
Are we running these tests anywhere? How much important are they to update?

cc: @openedx/axim-aximprovements

kdmccormick commented:
Ah, great question @farhan . Those tests are unfortunately skipped because they broke a long time ago (silently) and we weren't able to easily re-enable them. Let me take a look; it might be worth it to try to get these tests running for the sake of certainty about this PR.

kdmccormick commented:
Context:
* https://github.com/openedx/edx-platform/issues/35956
* https://github.com/openedx/edx-platform/pull/35955

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10048551618-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10048551618-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10048551618-tutor-requirements.txt)

abdullah8590 commented:
I verified the video xblock. The following are details for this.

Test execution Scenarios:
1. Validate that the video was added successfully to the studio
2. Validate the published video on LMS with the View live version button
3. Check all the options on the video Xblock
4. Check pause/play, full screen, and video speed

Issues:
* When the user deletes the video block in the studio, it shows an error at the bottom of the page.
<img width="1680" alt="image" src="https://github.com/user-attachments/assets/603007d7-1f12-4d12-b51b-3b99ed010521" />
* When the user sets the video duration in the studio, changes are not reflected in the live version. The user needs to publish the section, and then the changes appear on the LMS.

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10131359791-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10131359791-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10131359791-tutor-requirements.txt)

farhan commented:
> I verified the video xblock. The following are details for this.
> 
> Test execution Scenarios:
> 
> 1. Validate that the video was added successfully to the studio
> 2. Validate the published video on LMS with the View live version button
> 3. Check all the options on the video Xblock
> 4. Check pause/play, full screen, and video speed
> 
> Issues:
> 
> * When the user deletes the video block in the studio, it shows an error at the bottom of the page.
> 
> <img alt="image" width="1680" src="https://private-user-images.githubusercontent.com/185896520/447007927-603007d7-1f12-4d12-b51b-3b99ed010521.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgyNTg4NzEsIm5iZiI6MTc0ODI1ODU3MSwicGF0aCI6Ii8xODU4OTY1MjAvNDQ3MDA3OTI3LTYwMzAwN2Q3LTFmMTItNGQxMi1iNTFiLTNiOTllZDAxMDUyMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUyNlQxMTIyNTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMjhmMDAxM2JlNWJmMDZmYjM0YTg4MDM5MGFhMTkwNTVmMjQ1YzgzYjM0MDM5YTkxZmJjZDM0M2I1NWI3ZmY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.WcaH2QopySaGXgi1zYeV-n26tbpxmeo82_n6o80XrDg"> * When the user sets the video duration in the studio, changes are not reflected in the live version. The user needs to publish the section, and then the changes appear on the LMS.

Thanks for the testing @abdullah8590 

I have verified these issues exist on the master branch as well.

@kdmccormick perhaps we should go ahead and merge this?

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10151676421.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10151676421-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10151676421-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10151676421-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10154476684-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10154476684-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10154476684-tutor-requirements.txt)

kdmccormick commented:
@farhan , Dave Ormsbee expressed concern about merging this refactoring without any unit test coverage, as the VideoBlock JS is historically very issue-prone and hard to debug.

I've @ttqureshi to take a time-boxed [look into re-enabling the webpack JS tests](https://github.com/openedx/edx-platform/issues/35956). I suggested putting no more than 4 hours of work into finding a solution. If we cannot find a way to re-enable the tests in that amount of time, then I'm comfortable with accepting the risk and you moving forward with the merge whenever you are ready @farhan .

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10334915012.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10334915012-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10334915012-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10334915012-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10334914980-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10334914980-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10334914980-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10506024643.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10506024643-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10506024643-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10506024643-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10506024643.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10506024643-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10506024643-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10506024643-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10506058693.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10506058693-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10506058693-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10506058693-tutor-requirements.txt)

farhan commented:
I have converted this PR into draft because of the conflicts and changes happened in the following 2 files in the master branch:
1. ```xmodule/assets/video/public/js/08_video_auto_advance_control.js``` 
- https://github.com/openedx/edx-platform/pull/35200/files
2. ```xmodule/assets/video/public/js/09_video_caption.js``` 
- https://github.com/openedx/edx-platform/pull/36619/files
- https://github.com/openedx/edx-platform/pull/36732/files
- https://github.com/openedx/edx-platform/pull/36768/files

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10508947482.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10508947482-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10508947482-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10508947482-tutor-requirements.txt)

farhan committed:
refactor: Update and migrate Video Block JS files into xmodule/assets

- Move Video Block JS files from xmodule/js/src/video/ to xmodule/assets/video/public/js/
- Update JavaScript files from  RequireJS to ES6 import/export

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10726580464.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10726580464-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10726580464-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10726580464-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10731454876-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10731454876-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10731454876-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10760644375.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10760644375-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10760644375-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10760644375-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10766534948-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10766534948-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10766534948-tutor-requirements.txt)

ttqureshi commented on xmodule/assets/video/public/js/095_video_context_menu.js:
```diff
@@ -1,687 +1,699 @@
-                    this.getElement().removeClass('is-opened');
-                    return this;
-                },
 
```
please remove the empty line at the start of the file

ttqureshi reviewed (approved):
LGTM

ttqureshi commented:
TESTING NOTES:

- I've reviewed the code changes line-by-line and it seems fine to me.
- I've locally tested the JS changes and ti's working fine.
- Couldn't test it on the Sandbox, because of some issue with the sandbox.

farhan commented:
**Update in this PR:**
I have rebased the PR with master and found major conflicts mentioned in this [commit](https://github.com/openedx/edx-platform/pull/36595#issuecomment-3017244180). The conflicts have been resolved. This PR has been reviewed and tested by @ttqureshi and myself.

> @farhan , Dave Ormsbee expressed concern about merging this refactoring without any unit test coverage, as the VideoBlock JS is historically very issue-prone and hard to debug.
> 
> I've @ttqureshi to take a time-boxed https://github.com/openedx/edx-platform/issues/35956. I suggested putting no more than 4 hours of work into finding a solution. If we cannot find a way to re-enable the tests in that amount of time, then I'm comfortable with accepting the risk and you moving forward with the merge whenever you are ready @farhan .

@kdmccormick @feanil  — here is the time-boxed investigation by @ttqureshi: https://github.com/openedx/edx-platform/issues/35956#issuecomment-2969248222
- The investigation into re-enabling JS unit tests did not yield a quick path forward

@kdmccormick What's next, should we merge it? If yes please go ahead and merge it, announcing the stakeholders.
Let’s keep a close eye on post merge outcomes.
You probably have more context on the release schedule for the upcoming Open edX version.

cc: @openedx/axim-aximprovements

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10769062047.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10769062047-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10769062047-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10769062047-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10769062012-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10769062012-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10769062012-tutor-requirements.txt)

kdmccormick commented:
OK, thanks for the update @farhan , here is my proposed path forward:

* Ensure that https://github.com/openedx/xblocks-contrib/pull/37/ is updated with the latest JS code from this PR
* Get video JS tests running and passing on https://github.com/openedx/xblocks-contrib/pull/37/
* Trusting that the _extracted_ video JS tests are sufficient, merge this PR even though built-in video JS tests are not running.

Does that sound good?

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10785334527-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10785334527-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-10785334527-tutor-requirements.txt)

farhan commented:
> OK, thanks for the update @farhan , here is my proposed path forward:
> 
> * Ensure that [Extract video xblock xblocks-contrib#37](https://github.com/openedx/xblocks-contrib/pull/37) is updated with the latest JS code from this PR
> * Get video JS tests running and passing on [Extract video xblock xblocks-contrib#37](https://github.com/openedx/xblocks-contrib/pull/37)
> * Trusting that the _extracted_ video JS tests are sufficient, merge this PR even though built-in video JS tests are not running.
> 
> Does that sound good?

Yes, it sounds good

farhan commented:
**PR Update:**
PR merging is blocked, adding `blocked by other work` tag until the migrated JS pass the tests in the following PRs

https://github.com/openedx/edx-platform/pull/37343
https://github.com/openedx/xblocks-contrib/pull/37

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11306771920-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11306771920-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11306771920-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11356835267-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11356835267-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11356835267-tutor-requirements.txt)

salman2013 commented on xmodule/assets/video/public/js/03_video_player.js:
```diff
@@ -0,0 +1,911 @@
+    let time = this.videoPlayer.currentTime;
+
+    if (this.isFlashMode()) {
+        this.videoPlayer.currentTime = Time.convert(
```
This 'Time' object does not exist we need to  import it, this was failing during karma JS testing. 
I tried to fix the test by this [import](https://github.com/openedx/edx-platform/blob/9d71ce8151de5323c298e921b6d6c1ee7cbf72a6/xmodule/assets/video/public/js/04_video_control.js#L3) but its better if we can update this here.

salman2013 commented on xmodule/assets/video/public/js/04_video_control.js:
```diff
@@ -0,0 +1,164 @@
+    });
+}
+
+function updateVcrVidTime(params) {
```
This method's logic do not seems completely sync with the previous file in edx-platform
https://github.com/openedx/edx-platform/blob/a11086ffac05f8296937d63a4ecf387e5d450792/xmodule/js/src/video/04_video_control.js#L154

Is this intentional update?

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11385529257-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11385529257-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11385529257-tutor-requirements.txt)

salman2013 commented on xmodule/assets/video/public/js/09_events_plugin.js:
```diff
@@ -0,0 +1,177 @@
+    log: function(eventName, data) {
+        // use startTime and endTime to calculate the duration to handle the case where only a subsection of video is used
+        let endTime = this.state.config.endTime || this.state.duration;
+        let startTime = this.state.config.startTime;
```
We need to fix this like

`let startTime = this.state.config.startTime || 0; // Default to 0 if startTime is undefined
`

There is test in video_events_plugin_spec which test the scenario when startTime and endTime to be undefined, the code returning NaN in this case which should be fixed.

Test case:
https://github.com/salman2013/edx-platform/blob/8ecfb9bfa2723c015e3341491f37c84b8220405b/xmodule/js/spec/video/video_events_plugin_spec.js#L276

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11544140869-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11544140869-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11544140869-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11546859176-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11546859176-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11546859176-tutor-requirements.txt)

salman2013 committed:
test: Enable Karma Js tests (#37351)

* fix: enable and fix webpack karma js tests for video block

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11590346680.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11590346680-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11590346680-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11590346680-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11591092518-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11591092518-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11591092518-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11601575386-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11601575386-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11601575386-tutor-requirements.txt)

salman2013 reviewed (approved):
LGTM

farhan committed:
Merge branch 'master' into farhan/move-video-js-to-assets

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11616283034.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11616283034-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11616283034-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11616283034-tutor-requirements.txt)

kdmccormick commented on package.json:
```diff
@@ -15,7 +15,7 @@
     "test": "npm run test-jest && npm run test-karma",
     "test-jest": "jest",
-    "test-karma": "npm run test-karma-vanilla && npm run test-karma-require && echo 'WARNING: Skipped broken webpack tests. For details, see: https://github.com/openedx/edx-platform/issues/35956'",
+    "test-karma": "npm run test-karma-vanilla && npm run test-karma-require && npm run test-xmodule-webpack && echo 'WARNING: Skipped broken webpack tests. For details, see: https://github.com/openedx/edx-platform/issues/35956'",
```
let's make it clear that xmodule-webpack is running, even though the other two -webpack suites are not.
```suggestion
    "test-karma": "npm run test-karma-vanilla && npm run test-karma-require && npm run test-xmodule-webpack && echo 'WARNING: Skipped broken lms-webpack and cms-webpack tests. For details, see: https://github.com/openedx/edx-platform/issues/35956'",
```

kdmccormick reviewed (approved):
Awesome 🎉 

Before merging, can you confirm that you have manually tested both the CMS and LMS views of the video block?

farhan committed:
Update package.json

Co-authored-by: Kyle McCormick <kyle@axim.org>

farhan commented:
@kdmccormick 
Yes we have tested the PR manually.

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11619253108-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11619253108-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11619253108-tutor-requirements.txt)

farhan committed:
Merge branch 'master' into farhan/move-video-js-to-assets

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-36595-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-36595-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11632363191-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11632363191-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-36595-139931-51564656-11632363191-tutor-requirements.txt)
